### ISSUE TITLE
Inject papi-component style sheets

### ISSUE THREAD

**Issue by rolfheij-sil** [paranext/paranext-core#257]
Currently style sheets for papi-components are not packaged along with the actual components.
So when importing and using the components, the stylesheets should be imported separately.
We would like to package the style sheets along with the components, so that this separate importing is no longer needed.

Some information was gathered by @tjcouch-sil and shared on Discord:
```
Looks like this is a common problem with the way Vite is built. Vite doesn't do anything to automatically inject the styles, so we have to do it manually. Here are some links I found related to the issue:
https://stackoverflow.com/questions/76152013/react-component-library-built-with-vite-generates-one-big-style-css-how-to-spli
https://stackoverflow.com/questions/76488994/style-not-applying-to-imported-component-from-component-library
https://stackoverflow.com/questions/72555787/vite-vue-3-library-build-doesnt-implicitly-include-dist-style-css
https://github.com/vitejs/vite/issues/1579

This package seems like a pretty good solution, but it seems more complex to implement
https://www.npmjs.com/package/vite-plugin-lib-inject-css
```

Depends on #158 and #254

**Comment by tjcouch-sil** [comment]
Looks like we could do one of three things (at least; there are probably other options, but these seem reasonably clear at this point):

1. Use https://www.npmjs.com/package/@senojs/rollup-plugin-style-inject, a seemingly easy and quick way to have vite make code in our index file that injects our css into the document's head (may or may not support scss; I'm not sure)
2. Create our own node script based on https://www.npmjs.com/package/style-inject that runs as a build step and puts code into the generated index file to add the css after finishing building (supports scss)
3. Use https://www.npmjs.com/package/vite-plugin-lib-inject-css, which seems very complicated and may or may not support scss; I'm not sure

I lean toward 1 if we don't need to use scss right now or maybe 2 if we do. 1 essentially does 2 as part of the vite build process, which is nicer. 3 seems like out of scope for this current issue. I don't believe any of these things would be breaking changes for extension developers.

**Comment by rolfheij-sil** [comment]
Option1: When trying `npm run build:papi-component` I run into the following error:
```
failed to load config from C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts
error during build:
Error [ERR_REQUIRE_ESM]: require() of ES Module C:\Users\Paratext\Documents\Git\paranext-core\node_modules\@senojs\rollup-plugin-style-inject\index.js from C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts not supported.
Instead change the require of index.js in C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts to a dynamic import() which is available in all CommonJS modules.
    at _require.extensions.<computed> [as .js] (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64527:17)
    at Object.<anonymous> (C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts:39:49)
    at _require.extensions.<computed> [as .js] (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64524:24)
    at loadConfigFromBundledFile (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64532:21)
    at async loadConfigFromFile (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64388:28)
    at async resolveConfig (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:63993:28)
    at async build (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:46318:20)
    at async CAC.<anonymous> (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/cli.js:812:9)
```

**Comment by tjcouch-sil** [comment]
> Option1: When trying `npm run build:papi-component` I run into the following error:
> 
> ```
> failed to load config from C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts
> error during build:
> Error [ERR_REQUIRE_ESM]: require() of ES Module C:\Users\Paratext\Documents\Git\paranext-core\node_modules\@senojs\rollup-plugin-style-inject\index.js from C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts not supported.
> Instead change the require of index.js in C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts to a dynamic import() which is available in all CommonJS modules.
>     at _require.extensions.<computed> [as .js] (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64527:17)
>     at Object.<anonymous> (C:\Users\Paratext\Documents\Git\paranext-core\lib\papi-components\vite.config.ts:39:49)
>     at _require.extensions.<computed> [as .js] (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64524:24)
>     at loadConfigFromBundledFile (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64532:21)
>     at async loadConfigFromFile (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:64388:28)
>     at async resolveConfig (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:63993:28)
>     at async build (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:46318:20)
>     at async CAC.<anonymous> (file:///C:/Users/Paratext/Documents/Git/paranext-core/node_modules/vite/dist/node/cli.js:812:9)
> ```

Looks like something I don't know what to do about this just from the error, so we will probably need to look together sometime when I get back. Thanks!
